### PR TITLE
Travis improvements

### PR DESCRIPTION
**Pull request by Pentarctagon**
A few minor travis cleanups/improvements.

As far as LTO goes, it couldn't actually be enabled yet due to causing build timeouts and also the boost unit tests fail with c++14, but now the switch is there at least.

### CODE REVIEW

File: utils/travis/steps/install.sh [hunk 1]
```diff
@@ -8,12 +8,15 @@ export PLAY_TEST=true
...
 export BOOST_TEST=true
+export LTO=false
 
 if [ "$OPT" = "-O0" ]; then
     export EXTRA_FLAGS_RELEASE="-O0"
     export PLAY_TEST=false
     export MP_TEST=false
     export WML_TEST_TIME=20
+else
+    export LTO=false
```
Comment (jyrkive) [inline utils/travis/steps/install.sh]: Is this line here in order to allow us to enable LTO for optimized builds by changing it?

If so, please add a comment to explain it.

Comment (Pentarctagon) [reply utils/travis/steps/install.sh]: Added.

**Comment by Pentarctagon** [comment]
And, I suppose for reference, the segfault that the boost unit tests get can be seen here:
https://travis-ci.org/Pentarctagon/wesnoth/jobs/349672098#L1198

File: utils/travis/steps/install.sh [hunk 2]
```diff
@@ -8,12 +8,16 @@ export PLAY_TEST=true
...
 export BOOST_TEST=true
+export LTO=false
 
 if [ "$OPT" = "-O0" ]; then
     export EXTRA_FLAGS_RELEASE="-O0"
     export PLAY_TEST=false
     export MP_TEST=false
     export WML_TEST_TIME=20
+else
+# change to true to enable LTO on optimized, non-xcode builds
```
Comment (jyrkive) [inline utils/travis/steps/install.sh]: Please indent this line as much as the following line.

Comment (Pentarctagon) [reply utils/travis/steps/install.sh]: Done.

**Comment by jyrkive** [comment]
Merged. Thanks! :)

**Pull request merged**